packages/react-admin/src/mui/form/SanitizedForm.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { propTypes as formPropTypes } from 'redux-form';
+
+const sanitizeFormProps = ({
+    anyTouched,
+    asyncValidate,
+    asyncValidating,
+    clearSubmit,
+    dirty,
+    handleSubmit,
+    initialized,
+    initialValues,
+    pristine,
+    submitting,
+    submitFailed,
+    submitSucceeded,
+    valid,
+    pure,
+    triggerSubmit,
+    clearSubmitErrors,
+    clearAsyncError,
+    blur,
+    change,
+    destroy,
+    dispatch,
+    initialize,
+    reset,
+    invalid,
+    touch,
+    untouch,
+    validate,
+    save,
+    translate,
+    autofill,
+    submit,
+    resource,
+    redirect,
+    array,
+    ...rest
+}) => rest;
+
+const SanitizedForm = ({ renderLayout, ...props }) =>
+    renderLayout ? (
+        renderLayout({
+            ...props,
+            getFormProps: sanitizeFormProps,
```

Comment:
The name implies it is a function. Can you either make it a function or rename it `formProps` ? I think the second is better as a function does not bring anything here